tests/frame_format_tests.cpp

```diff
@@ -0,0 +1,442 @@
+#include <sstream>
+#include <fstream>
+#include <adm/utilities/object_creation.hpp>
+#include <catch2/catch.hpp>
+#include <adm/serial/frame_format.hpp>
+#include <adm/serial/frame_header.hpp>
+#include <adm/document.hpp>
+#include <adm/write.hpp>
+#include <adm/parse.hpp>
+#include <adm/serial/changed_ids.hpp>
+#include "helper/ostream_operators.hpp"
+
+using namespace adm;
+using namespace std::chrono_literals;
+TEST_CASE("Invalid format id parsing") {
+  CHECK_THROWS(parseFrameFormatId("FF_00000000"));  // bad value
+  CHECK_THROWS(parseFrameFormatId("FA_00000001"));  // bad prefix
+  CHECK_THROWS(parseFrameFormatId("FF_0000000"));  // too short
+  CHECK_THROWS(parseFrameFormatId("FF_000000001"));  // too long
+  CHECK_THROWS(parseFrameFormatId("FA_00000001_01"));  // bad prefix
+  CHECK_THROWS(parseFrameFormatId("FF_00000001_00"));  // bad value
+  CHECK_THROWS(parseFrameFormatId("FF_00000001001"));  // no divider
+  CHECK_THROWS(parseFrameFormatId("FF_00000001_0"));  // postfix too short
+  CHECK_THROWS(parseFrameFormatId("FF_00000001_"));  // postfix too short
+  CHECK_THROWS(parseFrameFormatId("FF_00000001_001"));  // postfix too long
+  CHECK_THROWS(parseFrameFormatId("FF_000000g1"));  // bad hex
+}
+
+TEST_CASE("Short format id parsing") {
+  CHECK(parseFrameFormatId("FF_00000001").get<FrameIndex>() == 1);
```

Comment:
i get signed/unsigned comparison warnings from these

Reply:
a5545afefbc758c08585331f53b0d5d211902a8f fixes this